AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py

```diff
@@ -101,30 +101,58 @@ def enter_game_mode(msgtuple_success_fail: Tuple[str, str]) -> None:
 
-        :return: None
+        :return: True if the message is found, otherwise false.
         """
+        for printInfo in print_infos:
+            if printInfo.window == window.strip() and printInfo.message.strip() == expected_message:
+                return True
+        return False
 
-        # looks for an expected line in a list of tracers lines
-        # lines: the tracer list of lines to search. options are section_tracer.warnings, section_tracer.errors, section_tracer.asserts, section_tracer.prints
-        # return: true if the line is found, otherwise false
-        def find_expected_line(expected_line, lines):
-            found_lines = [printInfo.message.strip() for printInfo in lines]
-            return expected_line in found_lines
+    @staticmethod
+    def wait_for_critical_expected_line(window, expected_message, print_infos, time_out):
+        """
+        Looks for an expected line in a list of tracer log lines for a period of time.
+        Reports a critical result based on if the expected line was found. The result is successful if the line is found.
+        :param window: The log's window name. For example, logs printed via script-canvas use the "Script" window. 
+        :param expected_message: The log message we're expecting to find.
+        :param print_infos: A list of PrintInfos collected by Tracer to search. Example options: your_tracer.warnings, your_tracer.errors, your_tracer.asserts, or your_tracer.prints 
+        :param time_out: The total amount of time to wait before giving up looking for the expected line.
 
-        def wait_for_critical_expected_line(expected_line, lines, time_out):
-            TestHelper.wait_for_condition(lambda : find_expected_line(expected_line, lines), time_out)
-            Report.critical_result(("Found expected line: " + expected_line, "Failed to find expected line: " + expected_line), find_expected_line(expected_line, lines))
+        :return: No return value, but if the expected message is found, a successful critical result is reported; otherwise failure.
+        """
+        TestHelper.wait_for_condition(lambda : TestHelper.find_expected_line(window, expected_message, print_infos), time_out)
+        Report.critical_result(("Found expected line: " + expected_message, "Failed to find expected line: " + expected_message), TestHelper.find_expected_line(window, expected_message, print_infos))
 
-        def wait_for_critical_unexpected_line(unexpected_line, lines, time_out):
-            TestHelper.wait_for_condition(lambda : find_expected_line(unexpected_line, lines), time_out)
-            Report.critical_result(("Unexpected line not found: " + unexpected_line, "Unexpected line found: " + unexpected_line), not find_expected_line(unexpected_line, lines))
+    @staticmethod
+    def wait_for_critical_unexpected_line(window, unexpected_line, print_infos, time_out):
```

Comment:
I would think about how you differentiate this from `wait_for_critical_expected_line`, they seem to be almost dupes of each other, both are calling `TestHelper.wait_for_condition(lambda : TestHelper.find_expected_line(window, unexpected_line | expected_message, print_infos), time_out)` under the hood. Its only how they check the condition for `Report.critical_result`, so they could be combined into one function but comments still seem to lack crisp definitions IMHO.

Minor: as already noted it seems to be an oxymoron to provide a list of unexpected content to wait for. Content cannot be unexpected if you are providing it. However this may be a naming convention we follow elsewhere in testing code, so happy to defer to convention.